test/common/services/security/test_user.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Role } from './role';
+import { User } from './user';
+import { FtrProviderContext } from '../../ftr_provider_context';
+import { Browser } from '../../../functional/services/browser';
+import { TestSubjects } from '../../../functional/services/test_subjects';
```

Comment:
`Browser` and `TestSubjects` are both part of the `functional` services, which in turn [depend on](https://github.com/elastic/kibana/blob/a1176b0e4a7e61449509fffd5bff9f9d40e6c42a/test/functional/services/index.ts#L20) these `common` services. We end up creating an indirect circular dependency here.

I feel like we should either:
1) Move this to `functional` services
2) Move `Browser` and `TestSubjects` to be part of the `common` services.

Option 2 is probably more feasible, but I don't know for sure. Removing this circular dependency would allow you to remove the `ts-ignore` statements below too, since the services would be properly available.

Reply:
I think there's a 3rd option we could consider.
  * Leave the core functionality of the test_user service in common/services (removing the browser and test_subject usage parts)
  * Create a new `functional_test_user` wrapper service in test/functional/services that calls the common test_user service and then does any browser/testSubject refresh type stuff.  

In fact, this cleans up some logic where the test_user service currently has to determine if there's a browser.

Reply:
@spalger had a lot of input on how this was implemented.  Maybe he can chime in here?

Reply:
@dmlemeshko and I discussed this issue and Dima even tried to change it, but in the end decided this was the least complicated solution for now.  If you're OK with this we'd like to go ahead and get this PR merged and we could look into this issue again afterwards.

Reply:
Lee's suggestion is more in line with the service design if you ask me, as moving the browser service to common requires moving many services that shouldn't be a part of the "common" configuration (because they're not base level services useful for configs like api_integration).

That said, I prefer this method to wrappers where all the condition logic is spread across two files. Conditionally loading the browser and testSubjects service allows enabling functionality when those services are in use and disabling the functionality when they're not in use, which seems totally logical to me. What don't you like about the implementation? The fact that `@ts-ignore` is necessary because `browser` and `testSubjects` services are not a part of the default provider context?

Reply:
> Lee's suggestion is more in line with the service design if you ask me, as moving the browser service to common requires moving many services that shouldn't be a part of the "common" configuration (because they're not base level services useful for configs like api_integration).
> 
> That said, I prefer this method to wrappers where all the condition logic is spread across two files. Conditionally loading the browser and testSubjects service allows enabling functionality when those services are in use and disabling the functionality when they're not in use, which seems totally logical to me. What don't you like about the implementation? The fact that `@ts-ignore` is necessary because `browser` and `testSubjects` services are not a part of the default provider context?

Yeah the `@ts-ignore`'s (specifically "service is not normally available in common") gave me the impression that we were doing something "wrong" here. I wasn't taking issue with the implementation so much as the organization. We have this service in `common`, which depends on `functional` services, which in turn depend on `common` services,

I don't have a strong enough opinion (or full understanding tbh) about these test services to hold this up though 🤷‍♂